security-provider-adapters/bouncycastle-adapter/src/main/java/de/adorsys/keymanagement/bouncycastle/adapter/services/persist/DefaultKeyStoreOperImpl.java

```diff
@@ -65,8 +79,12 @@ public String addToKeyStoreAndGetName(KeyStore ks, ProvidedKeyTemplate entry, Su
 
     @SneakyThrows
     private KeyStore generate(KeySet keySet, Supplier<char[]> defaultPassword, KeyMetadataOper useMetadataOper) {
-        KeyStore ks = KeyStore.getInstance("UBER"); // FIXME - BCFKS from provided load store config
-        ks.load(null);
+        KeyStore ks = KeyStore.getInstance(config.getType());
+        if ("BCFKS".equals(config.getType())) {
```

Comment:
It is better to have:
```
        KeyStore ks;
        ks.load(null);
        if ("BCFKS".equals(config.getType())) {
            ks = createBCFKSKeystore(config, ks);
        } else {
            ks = createKeyStoreByType(type);
        }
```
for readability

Reply:
What should be inside createKeyStoreByType then?
We can't ks.load before initialize ks.
I've renamed createBCFKSKeystore to loadBCFKSKeystore.
Is it good to explicitly return ks even if it changes as a parameter?

Reply:
Yes - createKeyStoreByType should return keystore
```
 KeyStore ks;
        if ("BCFKS".equals(config.getType())) {
            ks = createBCFKSKeystore(config);
        } else {
            ks = createKeyStoreByType(type);
        }
```

Reply:
of course without changeable parameters is better